app/components/views/Deposit/index.tsx

```diff
@@ -0,0 +1,334 @@
+import { cx } from "@emotion/css";
+import { ButtonPrimary, Text } from "@klimadao/lib/components";
+import { addresses } from "@klimadao/lib/constants";
+import { TCO2Icon } from "@klimadao/lib/resources";
+import { useWeb3 } from "@klimadao/lib/utils";
+import { Trans, t } from "@lingui/macro";
+import { AccountBalanceWalletOutlined } from "@mui/icons-material";
+import KeyboardArrowDown from "@mui/icons-material/KeyboardArrowDown";
+import {
+  approveDepositToken,
+  depositTokens,
+  getAllowance,
+} from "actions/deposit";
+import { BalancesCard } from "components/BalancesCard";
+import { CarbonTokenModal } from "components/CarbonTokenModal";
+import { DisclamerModal } from "components/DisclaimerModal";
+import { TransactionModal } from "components/TransactionModal";
+import { providers } from "ethers";
+import { formatEther } from "ethers-v6";
+import { tokenInfo } from "lib/getTokenInfo";
+import { CarbonToken, queryUserCarbonTokens } from "lib/queryUserCarbonTokens";
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { useSelector } from "react-redux";
+import { useAppDispatch } from "state";
+import { AppNotificationStatus, TxnStatus, setAppState } from "state/app";
+import { selectNotificationStatus } from "state/selectors";
+import * as styles from "./styles";
+
+interface Props {
+  address?: string;
+  isConnected: boolean;
+  provider?: providers.JsonRpcProvider;
+  onRPCError: () => void;
+  toggleModal: () => void;
+}
+
+export const Deposit = (props: Props) => {
+  const dispatch = useAppDispatch();
+  const { address, toggleModal } = useWeb3();
+  const [quantity, setQuantity] = useState("0.0");
+  const [showTransactionModal, setShowTransactionModal] = useState(false);
+
+  const [showModal, setShowModal] = useState(false);
+  const [allowance, setAllowance] = useState({ spender: "", value: "0" });
+  const [holdings, setHoldings] = useState<null | Array<CarbonToken>>(null);
+  const [selectedToken, setSelectedToken] =
+    useState<Pick<CarbonToken, "amount" | "token">>();
+
+  const setStatus = (statusType: TxnStatus | null, message?: string) => {
+    if (!statusType) return dispatch(setAppState({ notificationStatus: null }));
+    dispatch(setAppState({ notificationStatus: { statusType, message } }));
+  };
+
+  const formattedTokenBalance = selectedToken
+    ? formatEther(selectedToken?.amount?.toString())
+    : 0;
+
+  const insufficientTokens = Number(formattedTokenBalance) < Number(quantity);
+  const hasAllowance = !!Number(allowance.value);
+  const fullStatus: AppNotificationStatus | null = useSelector(
+    selectNotificationStatus
+  );
+  const status = fullStatus?.statusType;
+
+  useEffect(() => {
+    if (!address) return;
+    (async () => await getTokensHoldings())();
```

Comment:
```suggestion
    getTokensHoldings(); 
```

IMO, this should work just fine.